src/logic/claims/data.ts

```diff
@@ -98,7 +109,8 @@ export function useUserClaimById(claimId: string) {
       setChainData(
         'Loaded claim',
         updateImmutablyCurried((state) => {
```

Comment:
Retrospectively, I really dislike this `updateImmutablyCurried` name. 
Maybe at this point we might plug in some proper state management library like zustand/redux to help with performance.

WDYT?

Reply:
I'm not a fan of the `updateImmutablyCurried` name either, but I don't think we need to reach for a state management library. The app is simple enough. We can probably get rid of `updateImmutablyCurried` though and tweak the `setChainData()` API to:
```ts
setChainData(
  'Loaded policies',
  (state) => {
    state.policies.userPolicyIds = result.data.ids;
    state.policies.byId = { ...state.policies.byId, ...result.data.byId };
  }
);
```

Reply:
or simply call it `updateState` ?

Reply:
Hmm Immer uses `produce`, so I reckon we should probably follow their lead and call it `produceState`. My feeling is `setChainData` and `updateState` sounds too similar to each other.

Reply:
I am fine with `produceState` as well.

Reply:
FYI I will rename in https://linear.app/api3-dao/issue/DAO-214/rename-updateimmutablycurried-function